src/gui/qgsmaptoolcapture.h

```diff
@@ -320,6 +334,31 @@ class GUI_EXPORT QgsMapToolCapture : public QgsMapToolAdvancedDigitizing
     void stopCapturing();
 
   private:
+
+    /**
+     * Called when the geometry is captured
+     * \since QGIS 3.24
+     */
+    virtual void geometryCaptured( const QgsGeometry &geometry ) {Q_UNUSED( geometry )} SIP_FORCE
```

Comment:
I thought it was ok with forcing them with SIP_FORCE. Am I wrong?

Reply:
I'm not sure either way -- it just looks problematic to me when a protected method is guaranteed to work correctly

Reply:
quoting https://stackoverflow.com/a/2170696/1548052

> This lets the derived classes override the function to customize the behavior as needed, without further exposing the virtual functions directly by making them callable by derived classes (as would be possible if the functions were just protected). The point is that virtual functions exist to allow customization; unless they also need to be invoked directly from within derived classes' code, there's no need to ever make them anything

And if you consider the idea of the additional class to handle the avoidIntersections, we'll be dealing with the private handler on the geometry and it will be better to just not forward the specific handlers.